src/mon/OSDMonitor.cc

```diff
@@ -7925,10 +7925,22 @@ int OSDMonitor::prepare_command_pool_set(const cmdmap_t& cmdmap,
       ss << "error parsing integer value '" << val << "': " << interr;
       return -EINVAL;
     }
+    if (!g_conf()->mon_allow_pool_size_one) {
+      ss << "configuring pool without replicas is disabled by default; you must first set the mon_allow_pool_size_one config option to true before you can set pool size to 1";
```

Comment:
I actually think we don't want to reference the config option here. There's just no point when setting pool size 1 is a good idea and it's okay to make it hard for users to do dumb things. Tell them it's disabled by default and leave it at that.

Reply:
Also this text block output isn't covered by the check for "n == 1" so it'll show up whenever anybody tries to set any pool size; it needs to get moved?

Reply:
yep, missed that check! thanks!